code/core/src/main/java/com/adobe/marketing/mobile/services/ui/vnext/message/InAppMessageSettings.kt

```diff
@@ -0,0 +1,234 @@
+         */
+        fun dismissAnimation(dismissAnimation: MessageAnimation) =
+            apply { this.dismissAnimation = dismissAnimation }
+
+        /**
+         * Sets the color of the backdrop behind the message. This is the color behind the message when the message is taking over the UI.
+         * @param backgroundColor the hex color of the backdrop behind the message.
+         */
+        fun backgroundColor(backgroundColor: String) =
+            apply { this.backgroundColor = backgroundColor }
+
+        /**
+         * Sets the opacity of the backdrop behind the message. This is the opacity behind the message when the message is taking over the UI.
+         * @param backdropOpacity the opacity of the backdrop behind the message. This should be a value between 0.0f (transparent) and 1.0f(opaque).
+         */
+        fun backdropOpacity(backdropOpacity: Float) =
+            apply { this.backdropOpacity = backdropOpacity }
+
+        /**
+         * Sets the corner radius of the message.
+         * @param cornerRadius the corner radius of the message
+         */
+        fun cornerRadius(cornerRadius: Float) = apply { this.cornerRadius = cornerRadius }
+
+        /**
+         * Configures whether the message should take over the UI.
+         * @param shouldTakeOverUi whether the message should take over the UI. Should be set to true if the message should take over the UI, false otherwise.
+         */
+        fun shouldTakeOverUi(shouldTakeOverUi: Boolean) =
+            apply { this.shouldTakeOverUi = shouldTakeOverUi }
+
+        /**
+         * Sets the asset map for the message. This is a map of asset names to asset URLs.
+         * @param assetMap the asset map for the message
+         */
+        fun assetMap(assetMap: MutableMap<String, String>) =
+            apply { this.assetMap = assetMap }
+
+        /**
+         * Sets the gesture map for the message. This is a map of gesture names
+         * (as defined by [InAppMessageSettings.MessageGesture]'s) to gesture actions.
+         * @param gestureMap the gesture map for the message
+         */
+        fun gestureMap(gestureMap: Map<String, String>) =
+            apply {
+                for ((key, value) in gestureMap) {
+                    val gesture: MessageGesture? = MessageGesture.getGestureForName(key)
+                    gesture?.apply {
+                        this@Builder.gestures[gesture] = value
+                    }
+                }
+            }
+
+        fun build() = InAppMessageSettings(
+            content,
+            width,
+            height,
+            verticalInset,
+            horizontalInset,
+            verticalAlignment,
+            horizontalAlignment,
+            displayAnimation,
+            dismissAnimation,
+            backgroundColor,
+            backdropOpacity,
+            cornerRadius,
+            shouldTakeOverUi,
+            assetMap,
```

Comment:
Can you create a copy of `assetMap` and `gestures` so that it is not accidentally modified by the caller?